activerecord/lib/active_record/relation.rb

```diff
@@ -551,9 +552,20 @@ def touch_all(*names, time: nil)
     #
     # ==== Examples
     #
-    #   Person.where(age: 0..18).destroy_all
-    def destroy_all
-      records.each(&:destroy).tap { reset }
+    #   Person.where(age: 0..18).destroy_all(batch_size: 3000)
+    def destroy_all(batch_options = {})
+      if ActiveRecord::Base.destroy_all_in_batches
+        in_batches(**batch_options).each_record(&:destroy)
```

Comment:
This method should return `nil` now, or maybe `self`. And we probably need to call `reset`